dev/integration_tests/android_semantics_testing/lib/src/matcher.dart

```diff
@@ -155,6 +161,9 @@ class _AndroidSemanticsMatcher extends Matcher {
       return _failWithMessage('Expected isFocusable: $isFocusable', matchState);
     if (isFocused != null && isFocused != item.isFocused)
       return _failWithMessage('Expected isFocused: $isFocused', matchState);
+    // Heading is not available in all Android versions, so match anything if it is not set by the platform
+    if (isHeading != null && isHeading != item.isHeading && item.isHeading != null)
```

Comment:
Since this behavior is different than the other conditions I would call it out in the class documentation.